pyEpiabm/pyEpiabm/sweep/spatial_sweep.py

```diff
@@ -0,0 +1,93 @@
+#
+# Infection due to contact in between people in different cells
+#
+
+import random
+import numpy as np
+
+from pyEpiabm.core import Parameters
+from pyEpiabm.routine import SpatialInfection
+
+from .abstract_sweep import AbstractSweep
+
+
+class SpatialSweep(AbstractSweep):
+    """Class to run the inter-cell space infections
+    as part of the sweep function. Runs through cells
+    and calculates their infectiousness parameter and calculates
+    a poisson variable of how many people each cell should
+    infect. Then chooses other cells, and persons within that
+    cell to assign as infectee. Then tests a infection event
+    against each susceptible member of the place. The resulting
+    exposed person is added to an infection queue.
+    """
+
+    def __call__(self, time: float):
+        """
+        Given a population structure, loops over cells and generates
+        a random number of people to infect. Then decides which cells
+        the infectees should be found in and considers whether an
+        infection event occurs on individual and cell infectiousness
+        and susceptibility.
+
+        :param time: Current simulation time
+        :type time: int
+        """
+        timestep = int(time * Parameters.instance().time_steps_per_day)
+        # Double loop over the whole population, checking infectiousness
+        # status, and whether they are absent from their household.
+        for cell in self._population.cells:
+
+            # As this tracks intercell infections need to check number of
+            # cells is more than one (edge case but worth having)
+            if len(self._population.cells) == 1:
```

Comment:
Is there a way to have this outside the for loop, and just break the function call at that point? Would just save checking this for each cell, but it's not a big extra cost!